lib/puppet_x/puppetlabs/strings/yard/handlers/host_class_handler.rb

```diff
@@ -8,6 +8,21 @@ class PuppetX::PuppetLabs::Strings::YARD::Handlers::HostClassHandler < PuppetX::
         param_tuple << ( a[1].nil? ? nil : a[1].source )
       end
     end
+    tp = Puppet::Pops::Types::TypeParser.new
+    param_type_info = []
+    # Ensure that this pop object has parameters
+    if statement.pops_obj.respond_to? :parameters
```

Comment:
As I think about this again I'm wondering if this is truly necessary at all. I think I was being defensive since some puppet types have parameters and some do not, but this us clearly going to be a HostClassDefinition which always has parameters!